modules/plugins/languages/tex/build/builders/tectonic.nix

```diff
@@ -0,0 +1,203 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+} @ moduleInheritancePackage: let
+  # The name of the builder
+  name = "tectonic";
+
+  # The builder template
+  template = import ./builderTemplate.nix;
+
+  inherit (lib.options) mkOption mkEnableOption;
+  inherit (lib.types) enum ints listOf package str;
+  inherit (lib.nvim.config) mkBool;
+  inherit (builtins) concatLists elem map toString;
+
+  cfg = config.vim.languages.tex;
+in (
+  template {
+    inherit name moduleInheritancePackage;
+
+    options = {
+      enable = mkEnableOption "Whether to enable Tex Compilation Via Tectonic";
+
+      package = mkOption {
+        type = package;
+        default = pkgs.tectonic;
+        description = "tectonic package";
+      };
+
+      executable = mkOption {
+        type = str;
+        default = "tectonic";
+        description = "The executable name from the build package that will be used to build/compile the tex.";
+      };
+
+      # -- Flags --
+      keepIntermediates = mkBool false ''
+        Keep the intermediate files generated during processing.
+
+        If texlab is reporting build errors when there shouldn't be, disable this option.
+      '';
+      keepLogs = mkBool true ''
+        Keep the log files generated during processing.
+
+        Without the keepLogs flag, texlab won't be able to report compilation warnings.
+      '';
+      onlyCached = mkBool false "Use only resource files cached locally";
+      synctex = mkBool true "Generate SyncTeX data";
+      untrustedInput = mkBool false "Input is untrusted -- disable all known-insecure features";
+
+      # -- Options --
+      reruns = mkOption {
+        type = ints.unsigned;
+        default = 0;
+        example = 2;
+        description = ''
+          Rerun the TeX engine exactly this many times after the first.
+
+          Setting this value to 0 will disable setting this option.
+        '';
+      };
```

Comment:
It's better to prefix those with "Whether to ..." as boolean options. The resulting descriptions in the manual might be confusing as they are assertive rather than descriptive.